daemon/src/surface.rs

```diff
@@ -188,26 +207,31 @@ impl Surface {
                 }
             }
             loop {
-                let files: Vec<PathBuf> = WalkDir::new(path)
-                    .into_iter()
-                    .filter_map(|e| e.ok())
-                    .filter(|e| {
-                        if let Some(guess) = new_mime_guess::from_path(e.path()).first() {
-                            guess.type_() == "image"
-                        } else {
-                            false
+                let is_zero = self.current_index == 0;
+                let is_below_len =
+                    !self.image_paths.is_empty() && self.current_index < self.image_paths.len();
+
+                let img_path = match (is_zero, is_below_len) {
+                    // Use
+                    (false, true) => self.image_paths[self.current_index].clone(),
+                    (true, true) => self.image_paths[0].clone(),
```

Comment:
I think these two lines can be collapsed and that the match can be replaced with an `if is_below_len`.

Reply:
Ah of course, leftover from a previous approach I was taking not realising the index is unsigned (work in typescript way too much unfortunately)